numba/jitclass/_box.c

```diff
@@ -0,0 +1,186 @@
+/*
+Implements jitclass Box type in python c-api level.
+*/
+#include "_pymodule.h"
+
+typedef struct {
+    PyObject_HEAD
+    void *meminfoptr, *dataptr;
+} BoxObject;
+
+
+/* Store function defined in numba.runtime._nrt_python for use in box_dealloc.
+ * It points to a function is code segment that does not need user deallocation
+ * and does not disappear while the process is stil running.
+ */
+static void (*MemInfo_release)(void*) = NULL;
+
+
+/*
+ * Box.__init__()
+ * Takes no arguments.
+ * meminfoptr and dataptr are set to NULL.
+ */
+static
+int Box_init(BoxObject *self, PyObject *args, PyObject *kwds) {
+    static char *keywords[] = {NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "", keywords))
```

Comment:
I am using `PyArg_ParseTupleAndKeywords` to guard for invalid usage and since I can't mark a `initproc` to take no kwargs or even given no args.